crypto/http/http_client.c

```diff
@@ -1106,17 +1119,24 @@ int OSSL_HTTP_proxy_connect(BIO *bio, const char *server, const char *port,
     char *mbuf = OPENSSL_malloc(BUF_SIZE);
     char *mbufp;
     int read_len = 0;
-    int rv;
     int ret = 0;
     BIO *fbio = BIO_new(BIO_f_buffer());
+#ifndef OPENSSL_NO_SOCK
+    int rv;
```

Comment:
Better move the `int rv;` declaration inside the block guarded by `if (issocket)` since it is used only there.